modules/ardrivo/src/Servo.cxx

```diff
@@ -0,0 +1,56 @@
+/*
+ *  Servo.cxx
+ *  Copyright 2020 ItJustWorksTM
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#include "Arduino.h"
+#include "BoardDataDef.hxx"
+#include "Servo.h"
+
+constexpr int servo_min(int8_t min) { return MIN_PULSE_WIDTH - min * 4; }
+constexpr int servo_max(int8_t max) { return MAX_PULSE_WIDTH - max * 4; }
+constexpr uint8_t TRIM_DURATION = 2;
+
+uint8_t Servo::attach(int pin) { return this->attach(pin, MIN_PULSE_WIDTH, MAX_PULSE_WIDTH); }
+
+uint8_t Servo::attach(int pin, int minWidth, int maxWidth) {
+    if (board_info->pins_caps[pin].pwm_able) {
+        attached_pin = pin;
+        min = (MIN_PULSE_WIDTH - minWidth) / 4;
+        max = (MAX_PULSE_WIDTH - maxWidth) / 4;
+        is_attached = true;
+    }
+    return attached_pin;
+}
+
+void Servo::detach() { is_attached = false; }
+
+void Servo::write(int value) {
+    if (attached())
+        board_data->servo_value[attached_pin] = std::clamp(value, 0, 180);
+}
+
+void Servo::writeMicroseconds(int value) {
+    if (attached())
+        this->write(map(std::clamp(value, servo_min(min), servo_max(max)) - TRIM_DURATION, MIN_PULSE_WIDTH, MAX_PULSE_WIDTH, 0,
+                        180)); // Convert microseconds to degree
+}
+
+int Servo::read() { return attached() ? board_data->servo_value[attached_pin].load(std::memory_order_relaxed) : -1; }
```

Comment:
Use default memory order instead of manually specifying one

Reply:
order must be one of std::memory_order_relaxed, std::memory_order_consume, std::memory_order_acquire or std::memory_order_seq_cst. Otherwise the behavior is undefined. Which one is the default one?

Reply:
Dude, take a coffee. The parameter is defaulted, but you override the default; just don't specify the parameter and it will use the default

Reply:
[https://en.cppreference.com/w/cpp/atomic/atomic/load](url) It says here. That's why I said it.

Reply:
And at the top of the page you have the function declaration with a defaulted parameter, so I happily call bull-feces on you reading that page from top to bottom